### ISSUE TITLE
request: improve adp[[<vector>]] error msg

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1855]
Oce (for adp objects anyway -- I've not checked more) only permits one indexing at a time.  I am not inclined to change that behaviour, given that things have been this way for a long time without serious complaint, and I'm not even sure that it's useful, because the natural product is a list, and for that a user can just do e.g. `lapply(c("a", "b"), \(x) adp[[x]])`.  But I don't like the error message, which is a bit too opaque, so I'll fix that.

``` r
library(oce)
#> Loading required package: gsw
data(adp)
adp[[c("v","a")]]
#> Warning in if (i == "distance") {: the condition has length > 1 and only the
#> first element will be used
#> Warning in if (i %in% c("originalCoordinate", "oceCoordinate", "cellSize", : the
#> condition has length > 1 and only the first element will be used
#> Warning in if (i %in% metadataNames) x@metadata[[i]] else x@data[[i]]: the
#> condition has length > 1 and only the first element will be used
#> Error in x@data[[i]]: subscript out of bounds
```

<sup>Created on 2021-08-05 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by dankelley** [comment]
I've changed it so it gives as follows.  I'll keep this issue open until the test suite is passed (local 10min, external 30min).


``` r
library(oce)
#> Loading required package: gsw
data(adp)
adp[[c("v","a")]]
#> Error: In [[,adp-method() : may only extract 1 item at a time.
```

<sup>Created on 2021-08-05 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by dankelley** [comment]
Fixed in "main" branch, commit d7f891399b3d5a809b5fbdac7145771d01408ee1.  

When I was in there fixing this, I also found a typo from the other day (checking a `try` against `"try=error"` instead of against `"try-error"`).

I also changed some instances of `if (length(grep(...)))` to `if (grepl(...))`, which may be less confusing to readers.

It passed local and win-release checks, so I pushed to GH and am closing this.
